SmartThings: Expose the entity to control the AC unit beep

alexqzd opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Similar to #151404, some Samsung AC units have a beeper that can be turned on and off from the SmartThings API.
This PR adds a config switch to control it from Home Assistant if the feature is available on the device.

I used the name 'Beep' because that what Samsung uses, but I'm open to suggestions 
<img width="1290" height="796" alt="image" src="https://github.com/user-attachments/assets/1c07c247-0abd-4bde-8529-a507b33d1a1a" />


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

mnvr1991 commented:
It’s called ‘Sound Effect’ in the iOS SmartThings app. I’m not really sure what would be better, Beep or Sound Effect.

![IMG_3232](https://github.com/user-attachments/assets/04941373-412b-4540-94f2-6f70c96c8fa1)

CoMPaTech reviewed (changes requested):
Hi @alexqzd - nice addition - I think sound effect would be more clear also wrt the actual app unless you'd change it to be descriptive like 'Confirmation sound' or something likewise? Might be good to change the variable name any which way around? Additionally a small rebase is in order (json is sorted since another PR got merged).

Happy to approve for merge when you ready to decide between beep or something else - just setting to request changes to get a ping once it's ready. 👍

alexqzd committed:
SmartThings: Expose the entity to control the AC unit sound effect

alexqzd committed:
Rename beep to sound_effect for consistency

alexqzd committed:
Add snapshots for sound_effect switch entities

alexqzd commented:
@mnvr1991 Interesting! My SmartThings iOS app also shows "Beep", but I've renamed it to "Sound Effect" per the consensus. Also rebased to fix the JSON sorting.
<img width="645" height="1398" alt="IMG_9892" src="https://github.com/user-attachments/assets/72936a29-61aa-4979-a783-5df86b4d1518" />

alexqzd committed:
Fix snapshots: use sound_effect entity names with proper translations

alexqzd committed:
Remove unused beep and none entity snapshots

CoMPaTech reviewed (approved):
Nice addition and good cleanup as well, thanks @alexqzd
